soci/soci_index.go

```diff
@@ -334,23 +335,45 @@ func (b *IndexBuilder) Build(ctx context.Context, img images.Image) (*IndexWithM
 	}
 
 	sociLayersDesc := make([]*ocispec.Descriptor, len(manifest.Layers))
-	eg, ctx := errgroup.WithContext(ctx)
+	var wg sync.WaitGroup
+	var layerCount uint64
+	errChan := make(chan error, len(manifest.Layers))
+
 	for i, l := range manifest.Layers {
-		i, l := i, l
-		eg.Go(func() error {
+		wg.Add(1)
+		go func(i int, l ocispec.Descriptor) {
+			defer wg.Done()
 			desc, err := b.buildSociLayer(ctx, l)
 			if err != nil {
-				return fmt.Errorf("could not build zTOC for layer %s: %w", l.Digest.String(), err)
+				if err != errUnsupportedLayerFormat {
+					errChan <- err
+				}
```

Comment:
why do we only send (and print) non-`errUnsupportedLayerFormat` errors?

Reply:
`errUnsupportedLayerFormat` is no longer an actual error condition at this point; it should not be propagated or interrupt execution. It gets sent through the errChan to be printed so that we have potentially useful troubleshooting output.